modules/social_features/social_group/social_group.module

```diff
@@ -2998,3 +3011,90 @@ function social_group_views_pre_view(ViewExecutable $view, string $display_id, a
       );
   }
 }
+
+/**
+ * Implements hook_group_types().
+ *
+ * We currently only use this for nodes and users since they can be installed
+ * as group content.
+ */
+function social_group_group_types(): array {
+  return array_fill_keys(['node', 'user'], TRUE);
+}
+
+/**
+ * Implements hook_entity_base_field_info().
+ *
+ * Creates computed fields to get the groups related to an entity. There will be
+ * a field containing all groups. Each group type will also get a separate
+ * field.
+ */
+function social_group_entity_base_field_info(EntityTypeInterface $entity_type): array {
+  $fields = [];
+
+  $entity_types = \Drupal::moduleHandler()->invokeAll('group_types');
+  \Drupal::moduleHandler()->alter('group_types', $entity_types);
+
+  if (!isset($entity_types[$entity_type->id()])) {
+    return $fields;
+  }
+
+  $fields['groups'] = BaseFieldDefinition::create('entity_reference')
+    ->setName('groups')
+    ->setTargetEntityTypeId($entity_type->id())
+    ->setSetting('target_type', 'group')
+    ->setLabel(t('Get the related groups for this entity.'))
+    ->setDescription(t('A list of all the groups referencing this entity.'))
+    ->setTranslatable(FALSE)
+    ->setComputed(TRUE)
+    ->setCustomStorage(TRUE)
+    ->setCardinality(FieldStorageDefinitionInterface::CARDINALITY_UNLIMITED)
+    ->setClass(GroupGroupReferenceItemList::class)
+    ->setDisplayConfigurable('form', TRUE)
+    ->setDisplayOptions('form', [
+      'label' => 'hidden',
+      'region' => 'hidden',
+      'weight' => 50,
+    ])
+    ->setDisplayConfigurable('view', TRUE)
+    ->setDisplayOptions('view', [
+      'label' => 'hidden',
+      'region' => 'hidden',
+      'weight' => 50,
+    ]);
+
+  if (!$entity_types[$entity_type->id()]) {
+    return $fields;
+  }
+
```

Comment:
What about 
![Screenshot 2024-03-26 at 13 28 42](https://github.com/goalgorilla/open_social/assets/16667281/fecd89b2-b045-473a-a570-5047195ef7be)

seems duplicate?

Reply:
Yeah, looks like now those hooks can be merged